examples/async-critical/src/wasm.rs

```diff
@@ -0,0 +1,96 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! The program demonstrates asynchronous execution and
+//! how to use macros `gstd::async_init`/`gstd::async_main`.
+//!
+//!  `Init` method gets three addresses, sends "PING" messages
+//! to them and waits for at least two replies with any payload ("approvals").
+//!
+//! `Handle` processes only "PING" messages. When `handle` gets such message
+//! it sends empty requests to the three addresses and waits for just one approval.
+//! If an approval is obtained the method replies with "PONG".
+
```

Comment:
Add docs/description please, as per #2845

Reply:
It is test demo. Such demos will be moved their own directory in the future